sdk/program/src/stake/state.rs

```diff
@@ -342,28 +344,26 @@ impl Delegation {
     }
 
     pub fn stake(&self, epoch: Epoch, history: Option<&StakeHistory>) -> u64 {
-        self.stake_activating_and_deactivating(epoch, history).0
+        self.stake_activating_and_deactivating(epoch, history)
+            .effective
     }
 
-    // returned tuple is (effective, activating, deactivating) stake
     #[allow(clippy::comparison_chain)]
     pub fn stake_activating_and_deactivating(
         &self,
         target_epoch: Epoch,
         history: Option<&StakeHistory>,
-    ) -> (u64, u64, u64) {
-        let delegated_stake = self.stake;
-
+    ) -> StakeActivationStatus {
         // first, calculate an effective and activating stake
         let (effective_stake, activating_stake) = self.stake_and_activating(target_epoch, history);
 
         // then de-activate some portion if necessary
         if target_epoch < self.deactivation_epoch {
             // not deactivated
-            (effective_stake, activating_stake, 0)
+            StakeActivationStatus::activating(effective_stake, activating_stake)
         } else if target_epoch == self.deactivation_epoch {
             // can only deactivate what's activated
-            (effective_stake, 0, effective_stake.min(delegated_stake))
```

Comment:
Getting rid of the `min` is a big win.  I double-checked the logic to be sure, and it seems like there's no need for it.

Reply:
Thanks for double checking!